server/src/main/java/org/eclipse/lsp/cobol/service/delegates/references/ElementOccurrences.java

```diff
@@ -64,18 +67,23 @@
 
     private static Element findElementByPosition(CobolDocumentModel document, TextDocumentPositionParams position) {
         AnalysisResult result = document.getAnalysisResult();
-        return Stream.<Supplier<Optional<Element>>>of(
-            () -> findElementByPosition(result.getVariableDefinitions(), result.getVariableUsages(), position),
-            () -> findElementByPosition(result.getParagraphDefinitions(), result.getParagraphUsages(), position),
-            () -> findElementByPosition(result.getSectionDefinitions(), result.getSectionUsages(), position),
-            () -> findElementByPosition(result.getConstantDefinitions(), result.getConstantUsages(), position),
-            () -> findElementByPosition(result.getCopybookDefinitions(), result.getCopybookUsages(), position),
-            () -> findElementByPosition(result.getSubroutineDefinitions(), result.getSubroutineUsages(), position))
+        Optional<Element> fromTree = Optional.ofNullable(result.getRootNode())
+            .flatMap(rootNode -> findNodeByPosition(rootNode, position))
+            .filter(node -> node instanceof Context)
+            .map(Context.class::cast)
+            .map(ElementOccurrences::convertToElement)
+            .map(ElementOccurrences::constructElementsExcludingImplicits);
+        return fromTree.orElseGet(() -> Stream.<Supplier<Optional<Element>>>of(
+                () -> findElementByPosition(result.getParagraphDefinitions(), result.getParagraphUsages(), position),
```

Comment:
I think you may remove the paragrphDefinitions here because they are also a part of the tree.

Reply:
Well, sort of. Yes, we got paragraphs as nodes as part of our tree. But those nodes do not implement `Context` and can't give references and definitions. And I'm afraid of how many unit tests I need to fix if I'll remove `paragraphDefinitions` from `AnalysisResult`.

Reply:
Ok, that makes sense.